src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,95 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] storage = (T[]) new Object[DEFAULT_CAPACITY];
+    private int storageSize;
+
     @Override
     public void add(T value) {
-
+        add(value, storageSize);
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkStorageSize();
+        if (index < 0 || index > storageSize) {
+            throw new ArrayListIndexOutOfBoundsException("Insert index: "
+                    + index + "is not valid. Please try again ");
+        }
+        System.arraycopy(storage, index, storage, index + 1, storageSize - index);
```

Comment:
No need to perform arraycopy here, your storage is already grown and copied if needed using checkStorageSize() method

Reply:
Not fixed